libtransmission/utils.cc

```diff
@@ -193,6 +194,14 @@ bool tr_file_save(std::string_view filename, std::string_view contents, tr_error
     {
         return false;
     }
+#ifndef _WIN32
+    // set file mode per settings umask()
```

Comment:
shouldn't this just be done in `tr_sys_file_open_temp()` in file-posix rather than here?

Reply:
> shouldn't this just be done in `tr_sys_file_open_temp()` in file-posix rather than here?

https://github.com/transmission/transmission/pull/7195#pullrequestreview-2406903681

Reply:
This has been in *file-posix.cc* before.  It moved back to *utils.cc* at @tearfur's recommendation earlier in this PR conversation.  @tearfur wrote:

> I'm requesting this because we don't want _any_ temp files to have default permissions.

How would y'all like error handling to proceed?

Reply:
I started to make the exact same suggestion as @reardonia but @tearfur is right about this -- the mask makes sense for every file we want to save, but not necessarily for every call to `tr_sys_file_open_temp()`